src/celeritas/optical/model/AbsorptionInteractor.hh

```diff
@@ -0,0 +1,45 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/model/AbsorptionInteractor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "celeritas/optical/Interaction.hh"
+
+namespace celeritas
+{
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample optical absorption interaction.
+ *
+ * Absorption rate is governed by sampling its mean free path in the action
+ * loop. The interactor simply returns an interaction saying the optical
+ * photon has been absorbed.
+ */
+class AbsorptionInteractor
+{
+  public:
+    //! Sample an interaction (no RNG needed)
+    inline CELER_FUNCTION Interaction operator()() const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Sample an absorption interaction.
+ */
+CELER_FUNCTION Interaction AbsorptionInteractor::operator()() const
+{
+    return Interaction::from_absorption();
```

Comment:
The interaction result here will just be the proposed change to the photon track. We definitely need a way to say that the photon has been absorbed because the track will need to be killed. Whether or not that will perform a local energy deposition will be at a higher level where the `Interaction` result gets applied to each track. 

As for recording the actual energy deposit, I've currently written in the assumption that the energy of optical photons never changes, so when an absorption result is returned, the higher level loop should use the photon's current energy (if we decide to deposit it).

Reply:
Yes, we can assume that the energy of optical photons is constant along a step. As long as the photon energy is deposited somewhere in the higher level, it should be okay.  Extended to WLS, the energy is not conversed (as some of energy will be dissipated through phonons or heat during absorption and reemission), so the difference in energy may be deposited locally even though this delta energy is not currently deposited in Geant4.  So, I was thinking that any energy deposition may be handled inside physics interactors consistently (if we support fine detailed optical simulation inside layers of a fancy photo-tube, like SIPM).  Either way probably be okay for now.

Reply:
Thanks, that's good to know.

If we have access to the secondaries from the interaction, we should be able to calculate the energy deposition at a higher level, and so we don't have to check energy conservation across every interactor. I don't have a strong opinion either way at the moment.

We may also want to consider adding WLS now since it seems to have more complications to keep track of than Rayleigh or absorption.

Reply:
Good. It would be better to merge this MR first so that we move to integrate and move to the rest of the optical photon loop.  WLS implementation needs more brain-storming for how to support multiple time structures which is poorly designed in Geant4 (i.e., WLS, WLS2 which are just same codes, but with different reemission functions, which can be combined within one model).